cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,130 @@
+ * defaultPerm [defaultPermissions] 	- data regarding default permissions
+ * 	defaultPerm.file [string]					- default permissions for files
+ * 	defaultPerm.dir [string] 					- default permissions for directories
+ * 	defaultPerm.owner [string] 				- default owner
+ * 	defaultPerm.group [string] 				- default group
+ * search [searchSettings] 						- data regarding search options
+ * 	search.hidden [bool] 							- includes hidden files and directories in the search
+ * 	search.exclude [string] 					- files and/or directories to be excluded (regexp)
+ * 	search.baseDirectory [string]			- base directory
+ * level [int]                        - used to print the output in a hierarchical way
+ */
+func FindRecursive(path string, defaultPerm defaultPermissions, search searchSettings, verbose bool, level int) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// In verbose mode, we are printing the output in a hierarchical way
+	if verbose {
+		level++
+	}
+
+	for _, f := range files {
+		name := f.Name()
+
+		relativePath := strings.TrimPrefix(strings.Join([]string{path, name}, "/"), search.baseDirectory)
+		fullPath := strings.Join([]string{path, name}, "/")
+
+		if m, _ := regexp.MatchString(search.exclude, relativePath); m {
+			if verbose {
+				fmt.Printf("\x1b[33m%sExcluding %s\n\x1b[0m", strings.Repeat(" ", level), name)
+			}
+		} else {
+			if search.hidden || !strings.HasPrefix(name, ".") {
+				mode := f.Mode()
+				currentOwner, _ := user.LookupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Uid))
+				currentGroup, _ := user.LookupGroupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Gid))
+
+				currentPerm := currentPermissions{
+					permissions:     mode.Perm().String(),
+					owner:           currentOwner.Username,
+					group:           currentGroup.Name,
+					hasCorrectOwner: currentOwner.Username == defaultPerm.owner,
+					hasCorrectGroup: currentGroup.Name == defaultPerm.group,
+				}
+
+				if mode.IsRegular() {
+					currentPerm.defaultPermissions = defaultPerm.file
+					currentPerm.hasCorrectPermissions = checkPermissions(currentPerm.permissions, currentPerm.defaultPermissions)
+					printOutput(level, "f", fullPath, currentPerm, defaultPerm, verbose)
+				} else if mode.IsDir() {
+					currentPerm.defaultPermissions = defaultPerm.dir
+					currentPerm.hasCorrectPermissions = checkPermissions(currentPerm.permissions, currentPerm.defaultPermissions)
+					printOutput(level, "d", fullPath, currentPerm, defaultPerm, verbose)
+					FindRecursive(fullPath, defaultPerm, search, verbose, level)
+				}
+			}
+		}
+	}
+}
+
+// checkPermissions returns true if the permissions are correct (false in another case)
+/* currentPermissions [string] - permissions that the file or directory has (i.e. -rwxrwxr-x)
+ * defaultPermissions [string] - permissions that should have (i.e. rwxrwxr-x)
+ */
+func checkPermissions(currentPermissions, defaultPermissions string) bool {
```

Comment:
how does this style of comments render with `go doc` (or `godoc`)?
just try it locally.

Usually in Go the function documentation describes the parameters in a more "discursive" style. Please just take a look at some examples of the standard library.